diff --git a/receiver/prometheusreceiver/config.go b/receiver/prometheusreceiver/config.go
index 9fd31d7efc..728bb00eb8 100644
--- a/receiver/prometheusreceiver/config.go
+++ b/receiver/prometheusreceiver/config.go
@@ -16,7 +16,6 @@ import (
 	commonconfig "github.com/prometheus/common/config"
 	promconfig "github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/discovery"
-	"github.com/prometheus/prometheus/discovery/kubernetes"
 	"github.com/prometheus/prometheus/discovery/targetgroup"
 	"go.opentelemetry.io/collector/config/confighttp"
 	"go.opentelemetry.io/collector/config/configoptional"
@@ -125,14 +124,6 @@ func (cfg *PromConfig) Validate() error {
 			if err := validateHTTPClientConfig(&sc.HTTPClientConfig); err != nil {
 				return err
 			}
-
-			for _, c := range sc.ServiceDiscoveryConfigs {
-				if c, ok := c.(*kubernetes.SDConfig); ok {
-					if err := validateHTTPClientConfig(&c.HTTPClientConfig); err != nil {
-						return err
-					}
-				}
-			}
 		}
 	}
 
