diff --git a/receiver/prometheusreceiver/config.go b/receiver/prometheusreceiver/config.go
index 2a7d881423..e0931cc477 100644
--- a/receiver/prometheusreceiver/config.go
+++ b/receiver/prometheusreceiver/config.go
@@ -17,12 +17,10 @@ import (
 	commonconfig "github.com/prometheus/common/config"
 	promconfig "github.com/prometheus/prometheus/config"
 	"github.com/prometheus/prometheus/discovery"
-	"github.com/prometheus/prometheus/discovery/kubernetes"
 	"github.com/prometheus/prometheus/discovery/targetgroup"
 	"go.opentelemetry.io/collector/config/configoptional"
 	"go.opentelemetry.io/collector/confmap"
 
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/apiserver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/targetallocator"
 )
 
@@ -33,11 +31,6 @@ type Config struct {
 
 	TargetAllocator configoptional.Optional[targetallocator.Config] `mapstructure:"target_allocator"`
 
-	//  APIServer has the settings to enable the receiver to host the Prometheus API
-	// server in agent mode. This allows the user to call the endpoint to get
-	// the config, service discovery, and targets for debugging purposes.
-	APIServer apiserver.Config `mapstructure:"api_server"`
-
 	// ScrapeOnShutdown enables a final scrape before the receiver closes.
 	//
 	// NOTE: This final scrape ignores the configured scrape interval.
@@ -136,14 +129,6 @@ func (cfg *PromConfig) Validate() error {
 			if err := validateHTTPClientConfig(&sc.HTTPClientConfig); err != nil {
 				return err
 			}
-
-			for _, c := range sc.ServiceDiscoveryConfigs {
-				if c, ok := c.(*kubernetes.SDConfig); ok {
-					if err := validateHTTPClientConfig(&c.HTTPClientConfig); err != nil {
-						return err
-					}
-				}
-			}
 		}
 	}
 
diff --git a/receiver/prometheusreceiver/config.schema.yaml b/receiver/prometheusreceiver/config.schema.yaml
index c86ac659cd..f999fbfa25 100644
--- a/receiver/prometheusreceiver/config.schema.yaml
+++ b/receiver/prometheusreceiver/config.schema.yaml
@@ -698,9 +698,6 @@ $defs:
     type: object
 description: Config defines configuration for Prometheus receiver.
 properties:
-  api_server:
-    $ref: ./internal/apiserver.config
-    description: APIServer has the settings to enable the receiver to host the Prometheus API server in agent mode. This allows the user to call the endpoint to get the config, service discovery, and targets for debugging purposes.
   config:
     $ref: prom_config
     x-pointer: true
diff --git a/receiver/prometheusreceiver/factory.go b/receiver/prometheusreceiver/factory.go
index d70d5642be..509f1f5a6e 100644
--- a/receiver/prometheusreceiver/factory.go
+++ b/receiver/prometheusreceiver/factory.go
@@ -15,7 +15,6 @@ import (
 	"go.opentelemetry.io/collector/receiver"
 	"go.uber.org/zap"
 
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/apiserver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/metadata"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/targetallocator"
 )
@@ -41,7 +40,6 @@ func createDefaultConfig() component.Config {
 		TargetAllocator: configoptional.Default(targetallocator.Config{
 			ClientConfig: taClientConfig,
 		}),
-		APIServer: apiserver.DefaultConfig(),
 	}
 }
 
diff --git a/receiver/prometheusreceiver/metrics_receiver.go b/receiver/prometheusreceiver/metrics_receiver.go
index ce66e0a603..0c5c83b642 100644
--- a/receiver/prometheusreceiver/metrics_receiver.go
+++ b/receiver/prometheusreceiver/metrics_receiver.go
@@ -27,7 +27,6 @@ import (
 	"go.uber.org/zap/exp/zapslog"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/apiserver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/metadata"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/sharedpromconfig"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusreceiver/internal/targetallocator"
@@ -50,7 +49,6 @@ type pReceiver struct {
 	scrapeManager          *scrape.Manager
 	discoveryManager       *discovery.Manager
 	targetAllocatorManager *targetallocator.Manager
-	apiServerManager       *apiserver.Manager
 	registry               *prometheus.Registry
 	registerer             prometheus.Registerer
 	unregisterMetrics      func()
@@ -69,16 +67,6 @@ func newPrometheusReceiver(set receiver.Settings, cfg *Config, next consumer.Met
 		registry,
 	)
 	sharedCfg := sharedpromconfig.NewConfig(&baseCfg)
-	var apiServerManager *apiserver.Manager
-	if cfg.APIServer.Enabled {
-		apiServerManager = apiserver.NewManager(
-			set,
-			&cfg.APIServer,
-			sharedCfg,
-			registry,
-			registerer,
-		)
-	}
 
 	pr := &pReceiver{
 		cfg:          cfg,
@@ -92,7 +80,6 @@ func newPrometheusReceiver(set receiver.Settings, cfg *Config, next consumer.Met
 			cfg.TargetAllocator.Get(),
 			sharedCfg,
 		),
-		apiServerManager: apiServerManager,
 	}
 	return pr, nil
 }
@@ -124,13 +111,6 @@ func (r *pReceiver) start(ctx context.Context, host component.Host, opts prometh
 		close(r.configLoaded)
 	})
 
-	if r.apiServerManager != nil {
-		err := r.apiServerManager.Start(ctx, host, r.scrapeManager)
-		if err != nil {
-			r.settings.Logger.Error("Failed to start APIServer", zap.Error(err))
-		}
-	}
-
 	return nil
 }
 
@@ -256,11 +236,6 @@ func (r *pReceiver) Shutdown(ctx context.Context) error {
 	if r.unregisterMetrics != nil {
 		r.unregisterMetrics()
 	}
-	if r.apiServerManager != nil {
-		if err := r.apiServerManager.Shutdown(ctx); err != nil {
-			return err
-		}
-	}
 	return nil
 }
 
